### ISSUE TITLE
Plugins via composer

### ISSUE THREAD

**Issue by landsman** [copona/copona#146]
I was found this package: https://github.com/sasedev/composer-plugin-filecopier and solved thanks it moving files from my vendors to opencart structure. 

I think that can be good way, maybe symlink can be better...
Example: https://github.com/BurdaPraha/oc_tracy#installation

What do you think about it?

**Comment by prhost** [comment]
Composer can create custom types like installers, allowing you to define what to do with the package after installing. I did something a while ago, but for now it's slow, I've discovered that there are a lot of things to change before it works.

https://github.com/copona/composer-installers

**Comment by stefanhuber** [comment]
I think it is a good idea. However coping files might not be enough.

I solved some of the other aspects: https://github.com/beyondit/opencart-extension-installer

e.g. the opencart module allows php,xml or sql installers. the above support xml and php...

**Comment by landsman** [comment]
@stefanhuber wow, cool! I was think about some install script for future. This looks great! Is there support vQmod?

**Comment by arnisjuraga** [comment]
@stefanhuber @landsman - can discuss it more in details in Slack?

**Comment by prhost** [comment]
@stefanhuber The only problem is that `mapping` needs to define all the files in the extension :/

**Comment by landsman** [comment]
@prhost @stefanhuber rly? I want some recursively tool. Do you think about symlinks?

**Comment by arnisjuraga** [comment]
Symlinks is not supported on Windows. And more administrative overhead, I think.

**Comment by prhost** [comment]
Has another way I did in the beginning before modularizing my project, which would keep the extension as package inside the vendor of the composer, create a cli that will copy these files and send them to their paths, without deleting the package inside the vendor , Using the same package as a "map" to remove the files from the project, but I do not think it is good to use a remove operation automatically within the project, you risk deleting something that should not.

**Comment by RaidoDev** [comment]
Listen, there is no need to copy module files to Copona core. They _must_ be added in `/extensions/Vendor/Module/[admin|catalog]/...` structure. You are talking about that, right? 
 
Copona is already loading these files (not for all controllers, but will).

**Comment by landsman** [comment]
And what about some autoloading from composer, like at system with DI ?

**Comment by prhost** [comment]
I do not think so, because if you do not just use the installer I made for it, it will install, update and remove all extensions from the `extensions` folder

https://github.com/copona/composer-installers
